### PR TITLE
Switch to new SessionStorage / session restore API.

### PR DESCRIPTION
**Pull request by pocmo**
For upcoming AC changes: https://github.com/mozilla-mobile/android-components/pull/9250

The AC changes will land in 2021. :)

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/FenixApplication.kt [hunk 1]
```diff
@@ -159,6 +163,42 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
...
+        components.useCases.tabsUseCases.restore(sessionStorage)
+
+        // Now that we have restored our previous state (if there's one) let's setup auto saving the state while
+        // the app is used.
+        sessionStorage.autoSave(store)
+            .periodicallyInForeground(interval = 30, unit = TimeUnit.SECONDS)
+            .whenGoingToBackground()
+            .whenSessionsChange()
+
+        // TODO: Move into use case?
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/FenixApplication.kt:179]: Yes, I think we should move this. Had left a comment about this on the A-C PR too as I remembered adding this here.

File: app/src/main/java/org/mozilla/fenix/FenixApplication.kt [hunk 2]
```diff
@@ -159,6 +163,42 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
         components.appStartupTelemetry.onFenixApplicationOnCreate()
     }
 
+    private fun restoreBrowsingSession() = GlobalScope.launch(Dispatchers.Main) {
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/FenixApplication.kt:166]: nit: Naming suggestion in A-C PR, but let's keep consistent based on our decision.

File: app/src/main/java/org/mozilla/fenix/FenixApplication.kt [hunk 3]
```diff
@@ -159,6 +163,42 @@ open class FenixApplication : LocaleAwareApplication(), Provider {
...
+        sessionStorage.autoSave(store)
+            .periodicallyInForeground(interval = 30, unit = TimeUnit.SECONDS)
+            .whenGoingToBackground()
+            .whenSessionsChange()
+
+        // TODO: Move into use case?
+        store.dispatch(RestoreCompleteAction)
+    }
+
+    private fun removeTimedOutTabs() {
```
Comment (csadilek) [inline app/src/main/java/org/mozilla/fenix/FenixApplication.kt:183]: Let's file a follow-up to cover this with a test. I didn't have one before it seems ...

**Comment by pocmo** [comment]
Updated. And added latest AC Nightly version. This should compile now.

**Pull request merged**